synapse/dendrite/parser.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import synapse.telepath as s_telepath
+import synapse.lib.service as s_service
+
+logger = logging.getLogger(__name__)
+
+class Parser:
+    '''
+    Parser provides a base implementation for processing parsing jobs in Dendrite.
+
+    The Parser is intended to be used by sub-classes for the handling of basic operations required for processing jobs
+    in Dendrite. Sub-classes must implement ``register`` to register a MIME type of interest with a specific queue
+    name. Some care should be taken to ensure that queue names do not collide on a given deployment.
+
+    Keyword Arguments:
+        svcbus (str):
+            URL for the SvcBus service that is used to locate the Coordinator and Jobs services.
+
+        coordinator (str):
+            The name that the Coordinator service is shared as on the SvcBus.
+
+        jobs (str):
+            The name that the Jobs service is shared as on the SvcBus.
+
+    Raises:
+        TypeError:
+            If one of 'svcbus', 'coordinator', 'jobs' kwargs are not provided.
+    '''
+    _REQUIRED_ARGS = set(('svcbus', 'coordinator', 'jobs'))
+
+    def __init__(self, **kwargs):
+        self.opts = kwargs
+        self._assertOpts()
+        self.svcbus = s_telepath.openurl(self.opts.get('svcbus'))
+        self.proxy = s_service.SvcProxy(self.svcbus)
+        self.coordinator = self.proxy.getNameProxy(self.opts.get('coordinator'))
+        self.jobs = self.proxy.getNameProxy(self.opts.get('jobs'))
+
+    def register(self, mimeType, queue):
```

Comment:
Thinking through deployment and management, i'm not sure an individual worker should *ever* think he gets to register/unregister a mime type for a given queue.  This seems like ye olde "too much rope" issue.